### DISCUSSION TITLE
Multi-File with Hub Button

### DISCUSSION THREAD

**Discussion by pqhf5kd** [pybricks/support#1312] [General]
I've had success calling classes from other files using `import __ from __` and using the stop button example to change the hub light but I can't seem to get the stop button to call another class. Is it possible?

```
Disable the stop button.
hub.system.set_stop_button(None)

 Check the button for 5 seconds.
watch = StopWatch()
while watch.time() < 5000:

     Set light to green if pressed, else red.
    if hub.button.pressed():
        from remote import remote
    else:
        from shuttle import shuttle

 Enable the stop button again.
hub.system.set_stop_button(Button.CENTER)

 Now you can press the stop button as usual.
wait(5000)
```

When I run the program above, one of the classes runs instantly instead of waiting for a button press or timeout.

**Comment by laurensvalk** [comment]
In Python, a module runs when you import it, so could be what you're seeing. Pybricks works the same.

I'm not sure what the other modules look like, but you could do something like this:

`hello_world.py`:
```python

def say_hi():
    print("Hi there!")
```

And then in your main program:

```python
# Import the function, but don't run it yet
from hello_world import say_hi

# Run the function whenever you want
for i in range(4):
    say_hi()
```

**Comment by pqhf5kd** [comment]
Sorry, @laurensvalk, if 

> In Python, a module runs when you import it

Then what is stopping say_hi from running immediately in your example?

> ```
> # Import the function, but don't run it yet
> from hello_world import say_hi
> ```

**Reply by laurensvalk** [reply]
To see the difference, expand `hello_world.py` to something like this, and see what happens when you import it:

```python

print("Hello! This will print when the module is imported.")

# When the module is imported, the code below runs also, but just like when included in
# your main script, it only *defines* the function, it does not get *called*. You can choose
# where it gets called.
def say_hi():
    print("Hi there!")

```

It can appear a bit tricky at first, but when you realize it's not actually much different than running normal code, it gets really powerful and you can organize your code nicely.

**Reply by pqhf5kd** [reply]
Thank you. I had to modify my scripts because they didn't contain functions but now it's working :)
